client/blocks/data-center-picker/index.tsx

```diff
@@ -23,297 +16,148 @@ interface ExternalProps {
-							{ ! isRecommendedDataCenterPicked && (
-								<CompactPickerSelect
-									onChange={ ( event ) => onChange( event.currentTarget.value ) }
-									value={ value }
-								>
-									{ DataCenterOptions.map( ( option ) => (
-										<option key={ option.value } value={ option.value }>
-											{ option.label }
-										</option>
-									) ) }
-								</CompactPickerSelect>
-							) }
-						</PickerOption>
-					</CompactLabel>
-				</PickerOptions>
-			</div>
-		);
-	}
-
-	return (
-		<div>
-			{ ! isFormShowing && (
-				<div>
-					<span>
-						{ translate( 'Your site will be automatically placed in the optimal data center.' ) }
-					</span>
-					&nbsp;
-					<Button
-						variant="tertiary"
-						onClick={ () => {
-							onClickShowPicker();
-							setIsFormShowing( ! isFormShowing );
-						} }
-					>
-						{ translate( 'Choose a data center instead' ) }
-					</Button>
-				</div>
+				</FormHidden>
 			) }
 
 			{ isFormShowing && (
 				<Form>
-					<FormHeadingContainer>
-						<FormHeading>{ translate( 'Pick your primary data center' ) }</FormHeading>
-						<Button variant="tertiary" onClick={ onCancel }>
-							{ translate( 'Cancel' ) }
-						</Button>
-					</FormHeadingContainer>
-					<FormDescription>
-						{ translate(
-							'Choose a primary data center for your site. For redundancy, your site will replicate in real-time to a second data center in a different region. {{supportLink}}Learn more{{/supportLink}}.',
-							{
-								components: {
-									supportLink: (
-										<ExternalLink
-											icon
-											target="_blank"
-											href={ localizeUrl(
-												'https://wordpress.com/support/choose-your-sites-primary-data-center/'
-											) }
-										/>
-									),
-								},
-							}
-						) }
-					</FormDescription>
-					<div role="radiogroup">
-						<AutomaticFormLabel>
-							<FormLabelThumbnail>
-								<FormLabelThumbnailImg
-									src={ worldImg }
-									alt={ translate( 'Illustration of the world' ) }
-								/>
-							</FormLabelThumbnail>
-							<input
-								className="form-radio"
-								type="radio"
-								name="geo_affinity"
-								value=""
-								checked={ isRecommendedDataCenterPicked }
-								onChange={ () => onChange( RECOMMENDED_DATA_CENTER ) }
-							/>
-							<span>
-								{ translate( 'Automatically place my site in the optimal data center.' ) }
-							</span>
-						</AutomaticFormLabel>
-						<FormRadiosBar
-							isThumbnail
-							checked={ value }
-							onChange={ ( event ) => onChange( event.currentTarget.value ) }
-							items={ DataCenterOptions }
-							disabled={ false }
-						/>
-					</div>
+					<StyledSelectControl
+						label={ <StyledLabel>{ translate( 'Pick your primary data center' ) }</StyledLabel> }
+						help={
+							hasTranslation(
+								'For redundancy, your site will be replicated in real-time to another region.'
```

Comment:
@javierarce I added the shorter copy here with a fallback to the translated copy.